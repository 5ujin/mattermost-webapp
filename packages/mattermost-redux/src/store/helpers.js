// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

export const offlineConfig = {
    effect: (effect, action) => {
        if (typeof effect !== 'function') {
            throw new Error('Offline Action: effect must be a function.');
        } else if (!action.meta.offline.commit) {
            throw new Error('Offline Action: commit action must be preset.');
        }

        return effect();
    }
};
