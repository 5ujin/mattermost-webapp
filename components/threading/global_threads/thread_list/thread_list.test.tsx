// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React, {ComponentProps} from 'react';

import {mountWithIntl} from 'tests/helpers/intl-test-helper';

import ThreadList from './thread_list';

describe('components/threading/global_threads/thread_list', () => {
    let props: ComponentProps<typeof ThreadList>;

    beforeEach(() => {
        props = {
            currentFilter: '',
            someUnread: true,
            actions: {
                markAllRead: jest.fn(),
                setFilter: jest.fn(),
            },
        };
    });

    test('should match snapshot', () => {
        const wrapper = mountWithIntl(
            <ThreadList {...props}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should support markAllRead', () => {
        const wrapper = mountWithIntl(
            <ThreadList {...props}/>,
        );

        wrapper.find('.icon-playlist-check').simulate('click');
        expect(props.actions.markAllRead).toHaveBeenCalled();
    });

    test('should support filter:all', () => {
        const wrapper = mountWithIntl(
            <ThreadList {...props}/>,
        );

        wrapper.find('button').first().simulate('click');
        expect(props.actions.setFilter).toHaveBeenCalledWith('');
    });

    test('should support filter:unread', () => {
        const wrapper = mountWithIntl(
            <ThreadList {...props}/>,
        );

        wrapper.find('button .dot').simulate('click');
        expect(props.actions.setFilter).toHaveBeenCalledWith('unread');
    });
});

