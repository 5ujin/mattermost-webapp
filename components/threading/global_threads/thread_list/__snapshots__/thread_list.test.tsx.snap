// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/threading/global_threads/thread_list should match snapshot 1`] = `
<Memo(ThreadList)
  actions={
    Object {
      "markAllRead": [MockFunction],
      "setFilter": [MockFunction],
    }
  }
  currentFilter=""
  someUnread={true}
>
  <div
    className="ThreadList"
  >
    <div
      className="header"
    >
      <Memo(Button)
        className="Button___large Margined"
        isActive={true}
        onClick={[Function]}
      >
        <button
          className="Button Button___transparent is-active Button___large Margined"
          onClick={[Function]}
        >
          <span
            className="Button_label"
          >
            <FormattedMessage
              defaultMessage="All your threads"
              id="threading.filters.allThreads"
            >
              <span>
                All your threads
              </span>
            </FormattedMessage>
          </span>
        </button>
      </Memo(Button)>
      <Memo(Button)
        className="Button___large Margined"
        hasDot={true}
        isActive={false}
        onClick={[Function]}
      >
        <button
          className="Button Button___transparent Button___large Margined"
          onClick={[Function]}
        >
          <span
            className="Button_label"
          >
            <FormattedMessage
              defaultMessage="Unreads"
              id="threading.filters.unreads"
            >
              <span>
                Unreads
              </span>
            </FormattedMessage>
            <span
              className="dot"
            />
          </span>
        </button>
      </Memo(Button)>
      <div
        className="right-anchor"
      >
        <Memo(SimpleTooltip)
          content="Mark all as read"
          id="threadListMarkRead"
        >
          <OverlayTrigger
            defaultOverlayShown={false}
            delayShow={500}
            overlay={
              <Tooltip
                bsClass="tooltip"
                className="hidden-xs"
                id="threadListMarkRead"
                placement="top"
              >
                Mark all as read
              </Tooltip>
            }
            placement="top"
            trigger={
              Array [
                "hover",
                "focus",
              ]
            }
          >
            <OverlayTrigger
              defaultOverlayShown={false}
              delayShow={500}
              overlay={
                <OverlayWrapper
                  bsClass="tooltip"
                  className="hidden-xs"
                  id="threadListMarkRead"
                  intl={
                    Object {
                      "defaultFormats": Object {},
                      "defaultLocale": "en",
                      "formatDate": [Function],
                      "formatDateToParts": [Function],
                      "formatDisplayName": [Function],
                      "formatList": [Function],
                      "formatMessage": [Function],
                      "formatNumber": [Function],
                      "formatNumberToParts": [Function],
                      "formatPlural": [Function],
                      "formatRelativeTime": [Function],
                      "formatTime": [Function],
                      "formatTimeToParts": [Function],
                      "formats": Object {},
                      "formatters": Object {
                        "getDateTimeFormat": [Function],
                        "getDisplayNames": [Function],
                        "getListFormat": [Function],
                        "getMessageFormat": [Function],
                        "getNumberFormat": [Function],
                        "getPluralRules": [Function],
                        "getRelativeTimeFormat": [Function],
                      },
                      "locale": "en",
                      "messages": Object {},
                      "onError": [Function],
                      "textComponent": "span",
                      "timeZone": "Etc/UTC",
                      "wrapRichTextChunksInFragment": undefined,
                    }
                  }
                  placement="top"
                >
                  Mark all as read
                </OverlayWrapper>
              }
              placement="top"
              trigger={
                Array [
                  "hover",
                  "focus",
                ]
              }
            >
              <Memo(Button)
                className="Button___large Button___icon"
                onBlur={[Function]}
                onClick={[MockFunction]}
                onFocus={[Function]}
                onMouseOut={[Function]}
                onMouseOver={[Function]}
              >
                <button
                  className="Button Button___transparent Button___large Button___icon"
                  onBlur={[Function]}
                  onClick={[MockFunction]}
                  onFocus={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                >
                  <span
                    className="Button_label"
                  >
                    <span
                      className="Icon"
                    >
                      <i
                        className="icon-playlist-check"
                      />
                    </span>
                  </span>
                </button>
              </Memo(Button)>
            </OverlayTrigger>
          </OverlayTrigger>
        </Memo(SimpleTooltip)>
      </div>
    </div>
    <div
      className="threads"
    />
  </div>
</Memo(ThreadList)>
`;
